### PR TITLE
Chore: Properly close resources on failure

### LINKED ISSUE #2236: [Bug] Some failing tests hang CI and don't stop the process [0.1]
# 🐛 Bug report

## Description

The test script might be capturing too eagerly when a test fails or something might be going on with `nyc` and `ava` when something fails. This seems to happen more often on macOS (😞) than other platforms.

[Example here](https://dev.azure.com/webhint/webhint/_build/results?buildId=3182&view=logs&jobId=790316ed-8aa2-568f-3859-3e923bc648be&taskId=4d28356e-2121-5462-4447-b94ded044ac4&lineStart=3431&lineEnd=3432&colStart=1&colEnd=1)

We should add a safegard in the test script that if running a command takes more than XX minutes it fails on top of trying to figure out what's going on too.

### PR DESCRIPTION
**Pull request by molant**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- ~~Added/Updated related documentation.~~
- ~~Added/Updated related tests.~~

## Short description of the change(s)

Fix #2236

### CODE REVIEW

File: packages/utils-tests-helpers/src/hint-runner.ts [hunk 1]
```diff
@@ -244,6 +249,14 @@ export const testHint = (hintId: string, hintTests: HintTest[], configs: { [key:
         } catch (e) {
             console.error(e);
 
+            if (server) {
+                server.stop();
```
Comment (antross) [inline packages/utils-tests-helpers/src/hint-runner.ts]: Should this also be an `await`?
```suggestion
                await server.stop();
```

File: packages/utils-tests-helpers/src/hint-runner.ts [hunk 2]
```diff
@@ -244,6 +249,14 @@ export const testHint = (hintId: string, hintTests: HintTest[], configs: { [key:
...
+            }
+
+            if (engine) {
+                await engine.close();
+            }
+
             return t.fail(`${hintTest.name} throwed an exception:\n${e.message}\n${e.stack}`);
```
Comment (antross) [inline packages/utils-tests-helpers/src/hint-runner.ts]: Should some version of `stopConnector` also be called here? Not sure if `hintTest.after` ever performs cleanup, but if it does it won't get run.

If so, perhaps we should just move all of the cleanup to a `finally` block.

Comment (molant) [reply packages/utils-tests-helpers/src/hint-runner.ts]: Not super concerned about not calling `hintTest.after`. The test has already failed so if there's ever a `test.after` action it doesn't matter that much.

File: packages/utils-tests-helpers/src/hint-runner.ts (hunk 1 above)
Comment (molant) [reply packages/utils-tests-helpers/src/hint-runner.ts]: Yup

File: packages/utils-tests-helpers/src/hint-runner.ts (hunk 2 above)
Comment (antross) [reply packages/utils-tests-helpers/src/hint-runner.ts]: ok

**Pull request merged**